lib/dialects/mssql/connection-manager.js

```diff
@@ -166,6 +169,15 @@ class ConnectionManager extends AbstractConnectionManager {
 
     return connection && connection.loggedIn;
   }
+
+  versionSupport() {
+    const version = _.get(this.sequelize, 'options.databaseVersion', 0);
+
+    if (semver.valid(version) &&
+      semver.lt(this.sequelize.options.databaseVersion, '11.0.0')) {
+      logger.warn('SQL Server 2008 and lower versions are not supported by Sequelize');
```

Comment:
I think we should be using the deprecation functions instead 😛

Reply:
I've tried to use deprecations. There is a catch, deprecations cannot be spied, even if they are wrapped in another function. I can only spy on methods(like those of Logger) with sinon.

Another issue are deprecation codes, should we use a new code for each  dialect and required version?

Reply:
What I did is just wrap noops in `util.deprecate`, see `deprecations.js`

Example:
```
exports.noBoolOperatorAliases = deprecate(noop, 'A boolean value was passed to options.operatorsAliases. This is a no-op with v5 and should be removed.', 'SEQUELIZE0004');
```

Just use an incremental number for the code.

Reply:
Yes, I tried something similar. The problem is that `require('.../deprecations')` exports a node module composed of functions, which cannot be spied on. Yesterday I tried to no avail

```js
// deprecations.js
exports.noUnsupportedDB = (dialect, minVer, version) => deprecate(noop, `(${dialect}) Database version ${version} is unsupported. Only ${minVer} and higher are supported.`, 'SEQUELIZE0005')();

// connection-manager.test.js
const depreSpy = spy(deprecations, 'noUnsupportedDBVersion')
```

But the spy didn't work. It seems spy can only spy on an instance method.

Reply:
Oh you can't spy on them because they are referenced directly 🤔